psh/test_ls.py

```diff
@@ -0,0 +1,152 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh ls command test
+#
+# Copyright 2021, 2022 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import psh.tools.psh as psh
+from psh.tools.common import CONTROL_CODE, OPTIONAL_CONTROL_CODE, create_testdir
+
+# acceptable separators: white spaces (wss) + CC, CC + wss, wss
+SEPARATOR_PATTERN = r'(' + CONTROL_CODE + r'|\s+)+'
```

Comment:
(\s+)+ is an example of Evil Regex: https://en.wikipedia.org/wiki/ReDoS#Exponential_backtracking